force-app/main/default/lwc/barcodeScannerExample/__tests__/barcodeScannerExample.test.js

```diff
@@ -0,0 +1,183 @@
+                'lightning-button'
+            );
+
+        expect(elementScanQRCodeButton).not.toBeNull();
+    });
+
+    it('navigates to the expected record view when a QR code is correctly scanned', async () => {
+        // Property record values to compare component output against
+        const NAV_TYPE = 'standard__recordPage';
+        const NAV_ACTION_NAME = 'view';
+        const NAV_RECORD_ID = '0031700000pJRRWAA4';
+
+        // Stub barcodeScanner availability to true
+        setBarcodeScannerAvailable();
+
+        // Create initial BarcodeScannerExample element and attach to virtual DOM
+        const elementBarcodeScannerExample = createElement(
+            'c-barcode-scanner-example',
+            { is: BarcodeScannerExample }
+        );
+        document.body.appendChild(elementBarcodeScannerExample);
+
+        // Mount `Scan QR Code` button and trigger scan of property record ID
+        const elementScanQRCodeButton =
+            elementBarcodeScannerExample.shadowRoot.querySelector(
+                'lightning-button'
+            );
+        elementScanQRCodeButton.click();
+
+        // Wait for async scan function to settle
+        await flushPromises();
+
+        // Get data the NavigationMixin was called with
+        const { pageReference } = getNavigateCalledWith();
+
+        // Confirm redirection to expected property record
+        expect(pageReference.type).toBe(NAV_TYPE);
+        expect(pageReference.attributes.actionName).toBe(NAV_ACTION_NAME);
+        expect(pageReference.attributes.recordId).toBe(NAV_RECORD_ID);
+    });
+
+    it('triggers an error toast notification when the user cancels the scan', async () => {
+        // Stub barcodeScanner as available
+        setBarcodeScannerAvailable();
+
+        // Mock user canceling the scan
+        setUserCanceledScan();
+
+        // Create initial BarcodeScannerExample element and attach to virtual DOM
+        const elementBarcodeScannerExample = createElement(
+            'c-barcode-scanner-example',
+            { is: BarcodeScannerExample }
+        );
+        document.body.appendChild(elementBarcodeScannerExample);
+
+        // Create initial BarcodeScannerExample element and attach to virtual DOM
+        const element = createElement('c-barcode-scanner-example', {
+            is: BarcodeScannerExample
+        });
+        document.body.appendChild(element);
+
+        // Mount `Scan QR Code` button and trigger scan of property record ID
+        const elementScanQRCodeButton =
+            elementBarcodeScannerExample.shadowRoot.querySelector(
+                'lightning-button'
+            );
+        elementScanQRCodeButton.click();
+
+        // Wait for element to mount
+        await flushPromises();
+
+        // Check that cancelation toast was triggered
+        const { title } = showToastEventCalledWith();
+        expect(title).toBe('Scanning Canceled');
+    });
```

Comment:
If the ask is to check that the toast is called with the expected values (2 different options), not sure what else to do here + second toast test besides check the stubbed text